tests/watcher.go

```diff
@@ -0,0 +1,126 @@
+package tests
+
+import (
+	"errors"
+	"k8s.io/client-go/tools/cache"
+	"sync/atomic"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+
+	sspv1alpha1 "kubevirt.io/ssp-operator/api/v1alpha1"
+)
+
+type SspWatch interface {
+	Stop()
+	Updates() <-chan *sspv1alpha1.SSP
+	Error() error
+}
+
+func StartWatch(listerWatcher cache.ListerWatcher) (SspWatch, error) {
+	listObj, err := listerWatcher.List(v1.ListOptions{ResourceVersion: ""})
+	if err != nil {
+		return nil, err
+	}
+
+	list, err := meta.ListAccessor(listObj)
+	if err != nil {
+		return nil, err
+	}
+
+	watch := &sspWatch{
+		listerWatcher: listerWatcher,
+		stopCh:        make(chan struct{}),
+		updateCh:      make(chan *sspv1alpha1.SSP),
+		lastVersion:   list.GetResourceVersion(),
+	}
+	go watch.startWatch()
+	return watch, nil
+}
+
+type sspWatch struct {
+	listerWatcher cache.ListerWatcher
+	stopCh        chan struct{}
+	updateCh      chan *sspv1alpha1.SSP
```

Comment:
We can denote this as a receive only channel

Reply:
Yes, but it is not needed. The test code uses only the `SspWatch` interface, which provides only the receive channel.